javav2/example_code/cloudfront/src/test/java/CloudFrontSigningTest.java

```diff
@@ -0,0 +1,251 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import com.example.cloudfront.CreateBucketPolicy;
+import com.example.cloudfront.CreateCannedPolicyRequest;
+import com.example.cloudfront.CreateCustomPolicyRequest;
+import com.example.cloudfront.CreateDistribution;
+import com.example.cloudfront.CreateKeyGroup;
+import com.example.cloudfront.CreateOriginAccessControl;
+import com.example.cloudfront.CreatePublicKey;
+import com.example.cloudfront.DeleteDistribution;
+import com.example.cloudfront.DeleteSigningResources;
+import com.example.cloudfront.SigningUtilities;
+import org.junit.jupiter.api.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
+import software.amazon.awssdk.http.*;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.cloudfront.CloudFrontClient;
+import software.amazon.awssdk.services.cloudfront.cookie.CookiesForCannedPolicy;
+import software.amazon.awssdk.services.cloudfront.cookie.CookiesForCustomPolicy;
+import software.amazon.awssdk.services.cloudfront.model.*;
+import software.amazon.awssdk.services.cloudfront.url.SignedUrl;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import java.io.*;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Properties;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@TestInstance(TestInstance.Lifecycle.PER_METHOD)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class CloudFrontSigningTest {
+    private static Logger logger = LoggerFactory.getLogger(CloudFrontSigningTest.class);
+
+    private static CloudFrontClient cloudFrontClient;
+    private static S3Client s3Client;
+    private static String bucketName;
+    private static String fileNameToUpload;
+    private static String distributionId;
+    private static String originAccessControlId;
+    private static String distributionDomainName;
+    private static String publicKeyFileName;
+    private static String publicKeyId;
+    private static String keyGroupId;
+    private static String bucketAccessPolicy;
+    private static ApacheHttpClient apacheHttpClient = (ApacheHttpClient) ApacheHttpClient.create();
+    private static String privateKeyFullPath;
+
+    @BeforeAll
+    static void setUp() throws IOException {
+
+        // Run tests on Real AWS resources.
+        cloudFrontClient = CloudFrontClient.builder()
+                .region(Region.AWS_GLOBAL)
```

Comment:
Does the Java linter really want 8 spaces here? It happens all over, but it seems weird. If that's intentional and standard, then I guess it's fine. But I would double check your linting settings regarding tabs for chained calls.

Reply:
All Java examples use ths format.